trade-finance-manager-ui/server/controllers/utilisation-reports/helpers/reconciliation-for-report-helper.ts

```diff
@@ -209,51 +216,87 @@ const mapPaymentToPaymentDetailsPaymentViewModel = (
+      if (!paymentsReceived) {
+        return payments;
+      }
+
+      return [...payments, ...paymentsReceived.map((payment) => ({ ...payment, dateReconciled }))];
+    },
+    [] as (Payment & { dateReconciled?: IsoDateTimeStamp })[],
+  );
+
   const paymentIdToAmountDataSortValueMap = getKeyToCurrencyAndAmountSortValueMap(
-    allPayments.map(({ id, amount, currency }) => ({ key: id, currency, amount })),
+    allPaymentsWithDateReconciled.map(({ id, amount, currency }) => ({ key: id, currency, amount })),
   );
-  const paymentIdToDateReceivedDataSortValueMap: { [key: number]: number } = getPaymentsSortedByDateReceived(allPayments).reduce(
-    (map, { id }, index) => ({ ...map, [id]: index }),
-    {},
+  const paymentIdToDateReceivedDataSortValueMap: { [key: number]: number } = getKeyToDateSortValueMap(
+    allPaymentsWithDateReconciled.map(({ id, dateReceived }) => ({ key: id, date: dateReceived })),
+  );
+  const paymentIdToDateReconciledDataSortValueMap = getKeyToDateSortValueMap(
+    allPaymentsWithDateReconciled.map(({ id, dateReconciled }) => ({ key: id, date: dateReconciled })),
   );
 
-  return feeRecordPaymentGroups.reduce((paymentDetails, { feeRecords, paymentsReceived, status: feeRecordPaymentGroupStatus }) => {
-    if (!paymentsReceived) {
-      return paymentDetails;
-    }
+  return feeRecordPaymentGroups.reduce(
+    (paymentDetails, { feeRecords, paymentsReceived, status: feeRecordPaymentGroupStatus, reconciledByUser, dateReconciled }) => {
+      if (!paymentsReceived) {
+        return paymentDetails;
+      }
 
-    const mappedFeeRecords = feeRecords.map(({ facilityId, exporter }) => ({ facilityId, exporter }));
-    return [
-      ...paymentDetails,
-      ...paymentsReceived.map((payment) => ({
-        feeRecordPaymentGroupStatus,
-        payment: mapPaymentToPaymentDetailsPaymentViewModel(
-          payment,
-          paymentIdToAmountDataSortValueMap[payment.id],
-          paymentIdToDateReceivedDataSortValueMap[payment.id],
-        ),
-        feeRecords: mappedFeeRecords,
-      })),
-    ];
-  }, [] as PaymentDetailsViewModel);
+      const mappedFeeRecords = feeRecords.map(({ facilityId, exporter }) => ({ facilityId, exporter }));
+      return [
+        ...paymentDetails,
+        ...paymentsReceived.map((payment) => ({
+          feeRecordPaymentGroupStatus,
+          payment: mapPaymentToPaymentDetailsPaymentViewModel(
+            payment,
+            paymentIdToAmountDataSortValueMap[payment.id],
+            paymentIdToDateReceivedDataSortValueMap[payment.id],
+          ),
+          feeRecords: mappedFeeRecords,
+          reconciledBy: getFormattedReconciledByUser(reconciledByUser),
+          dateReconciled: {
+            formattedDateReconciled: getFormattedDateReconciled(dateReconciled),
+            dataSortValue: paymentIdToDateReconciledDataSortValueMap[payment.id],
+          },
+        })),
+      ];
+    },
+    [] as PaymentDetailsViewModel,
+  );
```

Comment:
Can we add some documentation for the consts in this file please, there is a lot going on. Lines 254, 265, 267, 271, 275

I would refactor this in a future PR.